var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -289,36 +318,66 @@ def write_library_dirs(f, dirs):
                         )
                     )
 
+    def setup_dependent_build_environment(self, env, depspec):
+        self.setup_build_environment(env)
+
     def setup_build_environment(self, env):
         # Tell numpy which BLAS/LAPACK libraries we want to use.
         # https://github.com/numpy/numpy/pull/13132
         # https://numpy.org/devdocs/user/building.html#accelerated-blas-lapack-libraries
         spec = self.spec
 
         # https://numpy.org/devdocs/user/building.html#blas
-        if 'blas' not in spec:
-            blas = ''
-        elif spec['blas'].name == 'intel-mkl' or \
-                spec['blas'].name == 'intel-parallel-studio':
-            blas = 'mkl'
-        elif spec['blas'].name == 'blis':
-            blas = 'blis'
-        elif spec['blas'].name == 'openblas':
-            blas = 'openblas'
-        elif spec['blas'].name == 'atlas':
-            blas = 'atlas'
-        elif spec['blas'].name == 'veclibfort':
-            blas = 'accelerate'
-        else:
-            blas = 'blas'
+        # Also from https://github.com/numpy/numpy/blob/main/numpy/distutils/system_info.py:
+        # The order of finding the locations of resources is the following:
+        #  1. environment variable
+        #  2. section in site.cfg
+        #  3. DEFAULT section in site.cfg
+        #  4. System default search paths (see ``default_*`` variables below).
+        # Only the first complete match is returned.
+        #
+        # The way the detection works, it performs the above scan FOR
+        # EACH different type of lapack. That means that if ATLAS is
+        # installed in a default location, py-numpy will always prefer
+        # it since atlas appears higher in distutils search
+        # order. Adding these environment variables short circuits
+        # this search and gives you the correct behavior.
+        blas = ''
+        if 'blas' in spec:
```

Comment:
Actually, now that I'm looking at this again, we also have a separate section of LAPACK. I don't think it's possible to merge those. Should we just keep `blas`, `lapack`, and env var setting as separate groups?

Reply:
I'm fine doing it either way, just let me know!

Reply:
Let's go back to the way you had it before, that will make it easier to remove later. Sorry for making you do extra work!

Reply:
No worries, it makes the end result better.  

I put it back in, added the extra mkl lib, and put in all the comments so it's easier to explain why we're doing all these crazy things.